maven-plugins/enforcer-maven-plugin/src/main/java/io/helidon/build/maven/enforcer/inclusivenaming/InclusiveNamingRule.java

```diff
@@ -0,0 +1,239 @@
+                .filter(this::include)
+                .forEach(fr -> {
+                    count.incrementAndGet();
+                    processFile(fr, failures);
+                });
+
+        if (failures.size() == 0) {
+            Log.info("Inclusive naming processed " + count.get() + " files.");
+        } else {
+            Log.info("Inclusive naming processed " + count.get() + " files, found " + failures.size() + " errors");
+        }
+        return failures;
+    }
+
+    private void processFile(FileRequest fr, List<RuleFailure> errors) {
+        try (BufferedReader bufferedReader = Files.newBufferedReader(fr.path(), StandardCharsets.UTF_8)) {
+            String line;
+            int lineNum = 0;
+            List<Term> found = new LinkedList<>();
+            while ((line = bufferedReader.readLine()) != null) {
+                // we want to start from 1
+                lineNum++;
+                for (Term term : terms) {
+                    if (line.toLowerCase().contains(term.term)) {
+                        found.add(term);
+                    }
+                }
+                if (!found.isEmpty()) {
+                    errors.add(RuleFailure.create(fr, lineNum, "inclusive naming found: " + String.join(", ", found.toString())));
+                }
+                found.clear();
+            }
+
+        } catch (IOException e) {
+            throw new EnforcerException("Failed to process file for inclusive naming: " + fr.relativePath());
+        }
+    }
+
+    private boolean include(FileRequest fr) {
+        for (FileMatcher exclude : excludes) {
+            if (exclude.matches(fr)) {
+                return false;
+            }
+        }
+
+        for (FileMatcher include : includes) {
+            if (include.matches(fr)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Fluent API builder for {@link io.helidon.build.maven.enforcer.inclusivenaming.InclusiveNamingRule}.
+     */
+    public static class Builder {
+        private InlusiveNamingConfig inclusiveNamingConfig;
+
+        private Builder() {
+        }
+
+        /**
+         * Update builder from maven configuration.
+         *
+         * @param inclusiveNamingConfig maven config
+         * @return updated builder
+         */
+        public Builder config(InlusiveNamingConfig inclusiveNamingConfig) {
+            this.inclusiveNamingConfig = inclusiveNamingConfig;
+            return this;
+        }
+
+        /**
+         * Build a new instance of {@link io.helidon.build.maven.enforcer.inclusivenaming.InclusiveNamingRule}.
+         *
+         * @return new rule
+         */
+        public InclusiveNamingRule build() {
+            return new InclusiveNamingRule(this);
+        }
+
+        List<Term> terms() {
+            List<Term> terms = new ArrayList<>();
+            try (InputStream is = InclusiveNamingRule.class.getResourceAsStream("index.json")) {
```

Comment:
I think we should have a possibility to override the location of the json file with a custom file.
Also please rename it to something else (such as `inclusive-names.json`)

Reply:
Considering that it a json structure instead of a list of words, I thought some users could set incorrect files. But I guess overall it is better to give them the option.

Reply:
Done